### PR TITLE
fix(python, rust): Initialize validity for `GroupsProxy::Slice` windows

### LINKED ISSUE #15483: `filter` + `arg_max` + `over` producing non-deterministic junk values
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

df = pl.DataFrame({
    "group": ["A", "A", "B", "B", "C", "C", "C", "C", "C"],
    "value": [2, 1, 0, 3, 2, 3, 5, 4, 4]
})

df.with_columns(pl.col("value").filter(False).arg_max().over("group"))

# shape: (9, 2)
# ┌───────┬────────────┐
# │ group ┆ value      │
# │ ---   ┆ ---        │
# │ str   ┆ u32        │
# ╞═══════╪════════════╡
# │ A     ┆ null       │
# │ A     ┆ null       │
# │ B     ┆ null       │
# │ B     ┆ 1          │
# │ C     ┆ 560934760  │
# │ C     ┆ 1          │
# │ C     ┆ 1894190896 │
# │ C     ┆ 1          │
# │ C     ┆ null       │
# └───────┴────────────┘
```

### Log output

_No response_

### Issue description

`arg_max()` (and `arg_min()`) produce junk values when combined with `.filter()` and `.over()`

### Expected behavior

Return correct index.

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.18
Index type:           UInt32
Platform:             macOS-13.6.1-arm64-arm-64bit
Python:               3.12.2 (main, Feb  6 2024, 20:19:44) [Clang 15.0.0 (clang-1500.1.0.2.5)]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            <not installed>
fsspec:               <not installed>
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           <not installed>
nest_asyncio:         <not installed>
numpy:                1.26.4
openpyxl:             <not installed>
pandas:               2.2.1
pyarrow:              15.0.0
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```

</details>

### PR DESCRIPTION
**Pull request by rob-sil**
Closes #15483

In the case of a `GroupsProxy::Slice`, validity wasn't getting set if a group had all of its items filtered out. Instead, uninitialized values would be used, leading to a mix of nulls and garbage values.

The other variant of `GroupsProxy`, the `GroupsProxy::Idx`, doesn't seem to have a problem when groups are filtered out entirely. To be safe, I left the initialization code before the `match` so it affects both variants.

### CODE REVIEW

File: py-polars/tests/unit/operations/test_window.py [hunk 1]
```diff
@@ -398,6 +398,24 @@ def test_window_filtered_aggregation() -> None:
     assert_frame_equal(out, expected)
 
 
+def test_window_filtered_false() -> None:
```
Comment (ritchie46) [inline py-polars/tests/unit/operations/test_window.py:401]:
```suggestion
def test_window_filtered_false_15483() -> None:
```

File: crates/polars-lazy/src/physical_plan/expressions/window.rs [hunk 2]
```diff
@@ -725,7 +725,7 @@ where
     } else {
         // We don't use a mutable bitmap as bits will have have race conditions!
         // A single byte might alias if we write from single threads.
-        let mut validity: Vec<bool> = Vec::with_capacity(len);
+        let mut validity: Vec<bool> = vec![false; len];
```
Comment (ritchie46) [inline crates/polars-lazy/src/physical_plan/expressions/window.rs:728]: L796 can now be removed.

**Comment by ritchie46** [comment]
Thank you for the fix @rob-sil.

**Comment by rob-sil** [comment]
Sorry, I forgot to run pre-commit!

**Pull request merged**